window/src/os/xdg_desktop_portal.rs

```diff
@@ -48,8 +66,52 @@ fn value_to_appearance(value: OwnedValue) -> anyhow::Result<Appearance> {
 }
 
 pub async fn get_appearance() -> anyhow::Result<Appearance> {
-    let value = read_setting("org.freedesktop.appearance", "color-scheme").await?;
-    value_to_appearance(value)
+    let mut state = STATE.lock().unwrap();
+    if state.appearance.is_some()
+        && (state.subscribe_running || state.last_update.elapsed().as_secs() < 1)
+    {
+        Ok(state.appearance.unwrap())
+    } else {
+        let value = read_setting("org.freedesktop.appearance", "color-scheme").await?;
+        match value_to_appearance(value) {
+            Ok(a) => {
+                state.appearance = Some(a);
+                state.last_update = Instant::now();
+                Ok(a)
+            }
+            err => err,
+        }
+    }
```

Comment:
```suggestion
        let appearance = value_to_appearance(value)?;
        state.appearance = Some(appearance);
        state.last_update = Instant::now();
        Ok(appearance)
    }
```

prefer to have less nesting/braces!

Reply:
Thanks, the `?` is indeed much more elegant here.